connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/ProcessingContext.java

```diff
@@ -216,4 +217,18 @@ public void reporters(Collection<ErrorReporter> reporters) {
         this.reporters = reporters;
     }
 
+    @Override
+    public void close() {
+        ConnectException e = new ConnectException("Failed to close all reporters");
+        for (ErrorReporter reporter : reporters) {
+            try {
+                reporter.close();
+            } catch (Throwable t) {
+                e.addSuppressed(t);
+            }
+        }
+        if (e.getSuppressed().length > 0) {
+            throw e;
+        }
```

Comment:
Regardless how high performance we want this code to be, as a matter of principle we shouldn't create an exception (and pay the cost of filling its stacktrace during instantiation) if we don't need to throw it.

I sketched a modification just with a +1 line, but feel free to adjust to your style of preference.

```suggestion
        ConnectException e = null;
        for (ErrorReporter reporter : reporters) {
            try {
                reporter.close();
            } catch (Throwable t) {
                e = e == null ? new ConnectException("Failed to close all reporters") : e;
                e.addSuppressed(t);
            }
        }
        if (e != null) {
            throw e;
        }
```

Reply:
Here's a nice blog re: exception costs and when they occur: 
https://shipilev.net/blog/2014/exceptional-performance